### ISSUE TITLE
InnerTypeLast should handle ctors and static init blocks

### ISSUE THREAD

**Issue by pbludov** [checkstyle/checkstyle#8394]
https://checkstyle.sourceforge.io/config_design.html#InnerTypeLast
From the Check description:

>>  Checks nested (internal) classes/interfaces are declared at the bottom of the primary (top-level) class after all method and field declarations. 

It is a bit unclear whether **constructors** should be declared before an inner type. I think they should be.

$ javac TestClass.java
```
(no output)
```

$ cat config.xml
```xml
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
    "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="InnerTypeLast">
        </module>
    </module>
</module>
```

$ cat TestClass.java
```java
class TestClass1 {

    public class Inner { // Expected violation
    }

    public TestClass1(Object x) { // ctor
    }

}

class TestClass2 {

    public class Inner { // Expected violation
    }

    static { // static init block
    }

}
```

$ RUN_LOCALE="-Duser.language=en -Duser.country=US"
$ java $RUN_LOCALE -jar checkstyle-8.34-all.jar -c config.xml TestClass.java
```
Starting audit...
Audit done.
```

**Comment by rnveach** [comment]
I agree ctors should be included. I question static ctors and instance ctors without the method declaration, but I think it does make some sense they are included too.

**Comment by romani** [comment]
fix is merged.
